### PR TITLE
DOC: sparse.linalg: add note about complex matrices to `splu` and `spilu`

### LINKED ISSUE #8140: sparse LU decomposition does not solve with complex right-hand side
The `solve` method of the sparse LU-decomposition `splu` or `spilu` throws a `TypeError` if called with a `numpy.array` of type `numpy.complex`. I am actually using `spilu` for preconditioning a gmres-solver required to perform a linear solve in a non-symmetric generalized eigenvalue problem. The eigenvectors are complex and hence the right-hand side for the linear solve can be complex.

### Reproducing code example:
```
import numpy as np
from scipy.sparse import csr_matrix
import scipy.sparse.linalg as sp_sparse_la

A = csr_matrix([[2.,-1.],[-1.,2.]])
n = A.shape[0]

v_real = np.random.randn(n)
v_cmplx = np.random.randn(n) + 1.0J * np.random.randn(n)

luA = sp_sparse_la.splu(A)
x_real = luA.solve(v_real)
x_cmplx = luA.solve(v_cmplx)
```

### Error message:
```
Traceback (most recent call last):
  File "dump.py", line 20, in <module>
    x_cmplx = luA.solve(v_cmplx)
TypeError: Cannot cast array data from dtype('complex128') to dtype('float64') according to the rule 'safe'
```

### Scipy/Numpy/Python version information:
```
('1.0.0', '1.13.3', sys.version_info(major=2, minor=7, micro=12, releaselevel='final', serial=0))
```

### PR DESCRIPTION
**Pull request by jgyasu**
#### Reference issue
Closes #8140

### CODE REVIEW

File: scipy/sparse/linalg/_dsolve/linsolve.py [hunk 1]
```diff
@@ -371,6 +371,10 @@ def splu(A, permc_spec=None, diag_pivot_thresh=None,
 
     Notes
     -----
+    When a real array is factorized and the returned SuperLU object ``solve()`` method
```
Comment (j-bowhay) [inline scipy/sparse/linalg/_dsolve/linsolve.py:374]:
```suggestion
    When a real array is factorized and the returned SuperLU object's ``solve()`` method 
```

File: scipy/sparse/linalg/_dsolve/linsolve.py [hunk 2]
```diff
@@ -371,6 +371,10 @@ def splu(A, permc_spec=None, diag_pivot_thresh=None,
 
     Notes
     -----
+    When a real array is factorized and the returned SuperLU object ``solve()`` method 
+    is used with complex arguments an error is generated. Instead cast the initial
```
Comment (j-bowhay) [inline scipy/sparse/linalg/_dsolve/linsolve.py:375]:
```suggestion
    is used with complex arguments an error is generated. Instead, cast the initial
```

File: scipy/sparse/linalg/_dsolve/linsolve.py [hunk 3]
```diff
@@ -371,6 +371,10 @@ def splu(A, permc_spec=None, diag_pivot_thresh=None,
...
+    matrix to complex and then factorize.
```
Comment (j-bowhay) [inline scipy/sparse/linalg/_dsolve/linsolve.py:376]:
```suggestion
    array to complex and then factorize.
```

**Review by j-bowhay — CHANGES_REQUESTED** [review]
Thanks @jgyasu ! I couple of suggested tweaks to the wording (I know I left the suggested text!) Please can you make the changes in both paragraphs

**Comment by jgyasu** [comment]
> Thanks @jgyasu ! I couple of suggested tweaks to the wording (I know I left the suggested text!) Please can you make the changes in both paragraphs

Thanks, I have made the changes. I should have checked for grammatical bits before :sweat_smile:

**Review by j-bowhay — APPROVED** [review]
Thanks for your first contribution @jgyasu keep them coming!

**Pull request merged**